### ISSUE TITLE
train a model with a new label

### ISSUE THREAD

**Issue by zp2546265641** [ultralytics/ultralytics#13316]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

hello, i just meet a problem. i don't know whether it is normal? when i start to train, then"Model summary: 225 layers, 11166947 parameters, 11166931 gradients, 28.8 GFLOPs

Transferred 349/355 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir runs/detect/train2', view at http://localhost:6006/
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v8.2.0/yolov8n.pt to 'yolov8n.pt'..."

it starts to download, but in my train.py, i have loaded a pretrained model.train.py  as follow:

#!/home/zp/anaconda3/envs/yolov8/bin/python3
# -*- coding: utf-8 -*-
from ultralytics import YOLO
import rospkg
rospack = rospkg.RosPack()
yolov8_ros_path = rospack.get_path('yolov8_ros') 
model = YOLO(f"{yolov8_ros_path}/weights/yolov8s_rotor.pt")  # load a pretrained model (recommended for training)
# Train the model with GPU
results = model.train(data=f"{yolov8_ros_path}/data/yolov8.yaml", epochs=100, imgsz=640, workers=16,device=0)

and i have another question, in my train/images, i just add the number 80th's images(the new class"rotor"), the dataset before i don't download. whether it will bring some problems? is it necessary to download all images to retrain or just use a retraines model with new images can also train a new model that can also detect all things including the classes before and the new i add.
wishes for your replying! thank you!

**Comment by zp2546265641** [comment]
oh！i just use yolov8s.pt and renamed "yolov8s_rotor.pt" as pretrained model, but when i use yolov8n.pt and renamed "yolov8n_rotor.pt", and then "TensorBoard: Start with 'tensorboard --logdir runs/detect/train', view at http://localhost:6006/
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v8.2.0/yolov8n.pt to 'yolov8n.pt'...
AMP: checks skipped ⚠️, offline and unable to download YOLOv8n. Setting 'amp=True'. If you experience zero-mAP or NaN losses you can disable AMP with amp=False.
train: Scanning /home/zp/rm_ws/src/dual_visual/Yolov8_ros/yolov8_ros/data/train/labels... 10 images, 0 backgrounds, 0 corrupt: 100%|██████████| 10/10 [00:00<00:00, 77.54it/s]"!

then it skip downloading!!! and so whatsi the problem? but when i train the twice, 

#!/home/zp/anaconda3/envs/yolov8/bin/python3
# -*- coding: utf-8 -*-
from ultralytics import YOLO
import rospkg
rospack = rospkg.RosPack()
yolov8_ros_path = rospack.get_path('yolov8_ros') 
model = YOLO(f"{yolov8_ros_path}/weights/yolov8n_rotor.pt")  # load a pretrained model (recommended for training)

# Train the model with GPU
results = model.train(data=f"{yolov8_ros_path}/data/yolov8.yaml", epochs=100, imgsz=640, workers=16,device=0)
it comes again:

TensorBoard: Start with 'tensorboard --logdir runs/detect/train', view at http://localhost:6006/
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v8.2.0/yolov8n.pt to 'yolov8n.pt'...
 10%|█████████████▊                                                                                                                            | 640k/6.23M [01:32<13:51, 7.07kB/s]
 and the downloading speed is so slow!

**Comment by zp2546265641** [comment]
i ask the gpt, it tells me to disable AMP with amp=False. and it will skip the check! i want to know the use of amp check? if i disable it? then what impact will it bring?

**Comment by zp2546265641** [comment]
oh!!!my goad!!! after my training, my best.pt now only can detect the rotor, other classes all miss! and maybe the sample is too less, only 10 .png, the detection effect is very poor, how can i improve my detection?

**Comment by glenn-jocher** [comment]
Hello!

It sounds like you're encountering a common issue when adding new classes to a pretrained model. Here are a few steps to improve your model's detection capabilities:

1. **Increase Dataset Size**: 10 images are quite minimal for training a robust model. Try to increase the number of images for the new class. More diverse and numerous training examples will help improve the model's ability to generalize.

2. **Rebalance Dataset**: Ensure that your dataset includes a balanced number of images for each class. An imbalance can lead the model to favor the overrepresented classes.

3. **Data Augmentation**: Implement data augmentation to artificially expand your training dataset. This includes transformations like rotation, scaling, flipping, and color adjustment.

4. **Fine-tuning**: Instead of training from scratch, continue training (fine-tuning) from a model that has already been trained on other classes. This helps the model retain knowledge learned from the original dataset.

5. **Regular Evaluation**: Frequently validate your model on a separate validation set to monitor its performance on unseen data. This helps in catching overfitting early.

6. **Hyperparameter Tuning**: Experiment with different learning rates, batch sizes, and other training parameters. Sometimes minor adjustments can lead to significant improvements.

By expanding your dataset and ensuring a balanced, diverse set of training examples, you should see an improvement in your model's performance. Keep iterating and good luck!

**Comment by zp2546265641** [comment]
> Hello!
> 
> It sounds like you're encountering a common issue when adding new classes to a pretrained model. Here are a few steps to improve your model's detection capabilities:
> 
> 1. **Increase Dataset Size**: 10 images are quite minimal for training a robust model. Try to increase the number of images for the new class. More diverse and numerous training examples will help improve the model's ability to generalize.
> 2. **Rebalance Dataset**: Ensure that your dataset includes a balanced number of images for each class. An imbalance can lead the model to favor the overrepresented classes.
> 3. **Data Augmentation**: Implement data augmentation to artificially expand your training dataset. This includes transformations like rotation, scaling, flipping, and color adjustment.
> 4. **Fine-tuning**: Instead of training from scratch, continue training (fine-tuning) from a model that has already been trained on other classes. This helps the model retain knowledge learned from the original dataset.
> 5. **Regular Evaluation**: Frequently validate your model on a separate validation set to monitor its performance on unseen data. This helps in catching overfitting early.
> 6. **Hyperparameter Tuning**: Experiment with different learning rates, batch sizes, and other training parameters. Sometimes minor adjustments can lead to significant improvements.
> 
> By expanding your dataset and ensuring a balanced, diverse set of training examples, you should see an improvement in your model's performance. Keep iterating and good luck!
nice to receive your reply! 
Normally, how many pictures are needed for a class to ensure a good effect? I use the original yolov8n.pt model and add the categories and corresponding images I need to resume training（config: resume: true）. Do I need to download the original dataset (which contains a lot of images and takes up a lot of local memory)?

**Comment by glenn-jocher** [comment]
@zp2546265641 hello!

Great questions! The number of images required for effective training can vary widely depending on the complexity of the objects and the variability in the images. Generally, a few hundred images per class are recommended for a robust model, but more can be beneficial, especially in complex scenarios.

Regarding your training setup, using the `resume: true` configuration with the original `yolov8n.pt` and adding your categories is a good approach. You don't necessarily need to download the entire original dataset if you're continuing training with additional classes. Just ensure your added dataset is sufficiently representative of the conditions under which you expect the model to perform.

Keep focusing on gathering quality, varied data for your new classes, and you should see improvements. Good luck! 🚀
